mesonbuild/interpreter/interpreter.py

```diff
@@ -2753,12 +2753,24 @@ def extract_incdirs(self, kwargs, key: str = 'include_directories'):
         return result
 
     @typed_pos_args('include_directories', varargs=str)
-    @typed_kwargs('include_directories', KwargInfo('is_system', bool, default=False))
+    @typed_kwargs(
+        'include_directories',
+        KwargInfo('is_system', bool, default=False),
+        KwargInfo(
+            'use_type',
+            str,
+            default='src-and-build-relative',
+            since='1.2.0',
+            validator=in_set_validator(set(build.include_dirs_use_kwargs_to_enum.keys())),
```

Comment:
don't use `keys()` here. There are a really specific set of cases you need `.keys()`, but in general avoid it, because the default iteration is keys only and avoids an indirection.